### DISCUSSION TITLE
[k3s-upgrade] Cluster is stuck in "Cluster is being upgraded" after successful Upgrade

### DISCUSSION THREAD

**Discussion by theAkito** [k3s-io/k3s#6713] [Q&A] [answered]
**Environmental Info:**
K3s Version: 

Between `v1.19.1+k3s1` and `v1.20.x+k3s1`.

Node(s) CPU architecture, OS, and Version: 

Ubuntu 20.04

Cluster Configuration:

1 Server

**Describe the bug:**

When upgrading via a Master Plan, the Cluster is stuck in "upgrading".

**Steps To Reproduce:**

Upgrade via a generic Master Plan.

**Expected behavior:**

Cluster is _not_ stuck in "upgrading".

**Actual behavior:**

Cluster is stuck in "upgrading".

**Additional context / logs:**

Was testing the [automatic upgrade toolkit](https://docs.k3s.io/upgrades/automated#install-the-system-upgrade-controller) on a Cluster several times & noticed, that at some point the Cluster went into "upgrading" state and didn't get out of it. Since then, I have performed more tests, i.e. more successful upgrades, but the state remains.

I previously tested even more & freely on a different cluster, yet it never got stuck in "upgrading" state.

To be clear about my goal: while this issue may be technically fixed in a newer version or there is some other type of fix available or it might be of interest to maintainers (which is why I opened this issue), I am not inquiring particularly about the "correct" k3s version - not only because we are upgrading anyway - my question is about how to fix this issue, when it appears.

How can I make the "upgrading" state disappear & make the cluster work normally again, after an automatic upgrade, that caused this issue?

____

https://github.com/k3s-io/k3s/issues/3007#issuecomment-789344170

This comment is curious, as it points to another big issue, of which I am not sure, whether it is fixed in newer versions.
The cluster complains, for example, that some directories or files cannot be mounted, however inside the jobs doing it, it is proven, that they are mounted & working. Perhaps the issue can be ignored, as it is only temporarily unavailable?

**Comment by brandond** [comment, marked as answer]
This appears to have been mostly caused by an issue in the kubelet that caused it to not properly track mounts (volumes, secrets, etc) across restarts of the kubelet. It seems to have been fixed or at least improved in more recent releases of Kubernetes, as I don't see too many folks complaining about it any more.

If you're asking about how to get the cluster to stop showing in the Rancher UI as "upgrading", I would probably ask in rancher/rancher, as this is not the right place for questions about that product.

**Reply by theAkito** [reply]
Okay, but if I'm updating from an old version, like 1.19.x, does the fix from the newer version work, or does the problem persist from the previous, pre-upgrade version?

**Reply by brandond** [reply]
The issue is in the kubelet, which is part of Kubernetes. If you upgrade K3s to a newer minor version where the issue has been addressed then you should stop experiencing it once the upgrade is complete.
